starters/basic-starter/pages/index.tsx

```diff
@@ -88,6 +90,7 @@ export default function IndexPage({ menus, events, places }: IndexPageProps) {
 export async function getStaticProps(
   context,
 ): Promise<GetStaticPropsResult<IndexPageProps>> {
+  await testApiCompatibility(ENTITY_TYPES, drupal);
```

Comment:
I think we should do this only in development mode since essentially this is a development tool.

I'm also wondering if we should do this in a middleware so we could do this for all pages with a single call 🤔

Reply:
I tried using Nextjs's `middleware()` but kept getting this error. 

`"Native Node.js APIs are not supported in the Edge Runtime" `

My guess is because the testApiCompatibility function uses the drupal client which uses `process`? 

But I added a workaround and instead added it to _document.tsx where it runs on the server side on every page.

Reply:
The custom document seems to to _exactly_ what want. However, there is one major caveat with that:

> To prepare for [React 18](https://nextjs.org/docs/advanced-features/react-18), we recommend avoiding customizing getInitialProps and renderPage, if possible.

https://nextjs.org/docs/advanced-features/custom-document#customizing-renderpage

I'm wondering if we need to move this back to the index even though it provides inferior UX. 😬

Reply:
Ah, I moved it back to index then.